content/6_Advanced/Eulerian_Tours.mdx

```diff
@@ -21,15 +21,143 @@ Mentioned in USACO Training ...
 
 ## Implementation
 
-<Resources>
-	<Resource
-		source="Benq (from KACTL)"
-		title="EulerPath"
-		url="https://github.com/bqi343/USACO/blob/07acbcfff3131445ecf89f781e80618b833cb592/Implementations/content/graphs%20(12)/DFS/EulerPath%20(12.2).h"
-	/>
-</Resources>
+### Mail Delivery
+
+First let's define what an **Eulerian** path is
+
+> An Eulerian path is a path that goes through every edge once.
+
+Similarly, an **Eulerian cycle** is an Eulerian path that starts and ends
+with the same node.
+
+An important condition is that a graph can be an Eulerian graph if and only if every node has an even degree.
```

Comment:
i thought a graph w/ 2 odd degree nodes also was possible?

Reply:
Nope. It's necessary for an Eulerian path that all noded have even degree.
https://en.wikipedia.org/wiki/Eulerian_path